src/core/language/service/language.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { CreateLanguageDto } from '../dto/create-language.dto';
+import { UpdateLanguageDto } from '../dto/update-language.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Language } from '../entities/language.entity';
+import { LANGUAGE_NOT_FOUND } from '../messages';
+
+@Injectable()
+export class LanguageService {
+  constructor(
+    @InjectRepository(Language)
+    private readonly languageRepository: Repository<Language>,
+  ) {}
+
+  async create(createLanguageDto: CreateLanguageDto): Promise<Language> {
+    const language = this.languageRepository.create(createLanguageDto);
+    await this.languageRepository.save(language);
+    return language;
+  }
+
+  async findAll(): Promise<Language[]> {
+    return this.languageRepository.find();
+  }
+
+  async findOne(id: number): Promise<Language> {
+    const language = await this.languageRepository.findOne({ where: { id } });
+    if (!language) {
+      throw new NotFoundException(LANGUAGE_NOT_FOUND);
+    }
+    return language;
+  }
+
+  async update(
+    id: number,
+    updateLanguageDto: UpdateLanguageDto,
+  ): Promise<Language> {
+    const language = await this.languageRepository.preload({
+      id,
+      ...updateLanguageDto,
+    });
+    if (!language) {
+      throw new NotFoundException(LANGUAGE_NOT_FOUND);
```

Comment:
Mas bien no?, por que hubo un problema al crearlo
```suggestion
      throw new NotFoundException(ERROR_ON_DATABASE_MESSAGE);
```